src/libponyc/pass/flatten.c

```diff
@@ -279,10 +279,12 @@ ast_result_t pass_flatten(ast_t** astp, pass_opt_t* options)
 
     case TK_FVAR:
     case TK_FLET:
-      return flatten_provides_list(options, ast, 3);
 
     case TK_EMBED:
     {
+      if(ast_childidx(ast, 3) == NULL)
```

Comment:
Unfortunately this isn't the right thing to do. With the change to the grammar, `embed` nodes will always have 3 childs and this condition will always be true, which means we'll never check that an embedded field has a valid type.

We should always go through the checks and return `AST_OK` instead of `flatten_provides_list` at the end.